ui/components/app/connected-accounts-permissions/connected-accounts-permissions.js

```diff
@@ -0,0 +1,83 @@
+import classnames from 'classnames';
+import PropTypes from 'prop-types';
+import React, { useState } from 'react';
+import CheckBox from '../../ui/check-box';
+import { useI18nContext } from '../../../hooks/useI18nContext';
+import { usePermissionDescriptions } from '../../../hooks/usePermissionDescriptions';
+
+const ConnectedAccountsPermissions = ({ permissions }) => {
+  const t = useI18nContext();
+  const [expanded, setExpanded] = useState(false);
+
+  const getPermissionDescription = usePermissionDescriptions();
+
+  const toggleExpanded = () => {
+    setExpanded(!expanded);
```

Comment:
When state changes in React, it effectively asynchronous. The `expanded` state here is the one from the most recent render, so this can result in the expanded state only toggling _once_ if you click it twice, for example.

Generally when making any state changes that depend upon the previous state value, like a toggle, you should pass a function into the setter that operates on the previous state value. This ensures the behavior is always what we'd expect, avoiding that race condition case.

```
    setExpanded((_expanded) => !_expanded);
```

Reply:
This makes sense. Thank you for the explanation!